tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1581,11 +1600,22 @@ void EnqueueRecordEventCommand::process() {
     HugepageDeviceCommand command_sequence(cmd_region, cmd_sequence_sizeB);
 
     CoreType dispatch_core_type = dispatch_core_manager::instance().get_dispatch_core_type(this->device->id());
-    uint32_t dispatch_message_addr = dispatch_constants::get(
+    uint32_t dispatch_message_base_addr = dispatch_constants::get(
         dispatch_core_type).get_device_command_queue_addr(CommandQueueDeviceAddrType::DISPATCH_MESSAGE);
 
+    uint32_t last_index = this->expected_num_workers_completed.size() - 1;
+    // We only need the write barrier for the last wait cmd
```

Comment:
Possible. I didn't commonize it since they use different fns `add_dispatch_wait` vs `add_dispatch_wait_with_prefetch_stall` and could potentially take different args, and wanted to avoid adding a wrapper fn where we'd need to match default args with the actual fns to keep same functionality.